### PR TITLE
ENH: Mesh-to-mesh image registration Python classes, tests, examples

### PR DESCRIPTION
**Pull request by tbirdso**
- Add Python implementation of mesh-to-mesh registration classes using ITK image registration techniques
- Add pytest modules to verify new registration classes
- Add example notebook to demonstrate new classes

### CODE REVIEW

**Comment by tbirdso** [comment]
This PR separates functional mesh-to-mesh registration classes (`MeanSquaresRegistrar` and `DiffeoRegistrar`) along with their tests and examples from ongoing mesh-to-mesh point set registration development in [PR 29](https://github.com/KitwareMedical/HASI/pull/29).

**Comment by tbirdso** [comment]
The failing checks `build-test-cxx (windows-2019)` and `build-macos-python-packages` are tracked in [Issue 30](https://github.com/KitwareMedical/HASI/issues/30) and appear to predate the proposed changes.

File: examples/MeshToMeshRegistration.ipynb [hunk 1]
```diff
@@ -0,0 +1,1150 @@
...
+    "DIFFEO_OUTPUT_FILE = 'Data/901-diffeo-registered.vtk'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Download meshes\n",
```
Comment (thewtex) [inline examples/MeshToMeshRegistration.ipynb:67]: Here we need:

```
os.makedirs('Data', exist_ok=True)
```

File: src/registration/meansquaresregistrar.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+# Filename: MeanSquaresRegistrar.py
+# Contributors: Tom Birdsong
+# Purpose: Python implementation of Slicer SALT mesh-to-mesh registration Cxx
+# CLI.
+#    Ported from https://github.com/slicersalt/RegistrationBasedCorrespondence/
+
+import logging
+import os
+import pytest
```
Comment (thewtex) [inline src/registration/meansquaresregistrar.py:11]: remove `import pytest`

File: src/registration/meshtomeshregistrar.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+# Filename: MeshToMeshRegistration
+# Contributors: Tom Birdsong
+# Purpose: Python implementation of Slicer SALT mesh-to-mesh registration Cxx
+# CLI.
+#    Ported from https://github.com/slicersalt/RegistrationBasedCorrespondence/
+import logging
+import os
+import pytest
```
Comment (thewtex) [inline src/registration/meshtomeshregistrar.py:10]: remove `import pytest`

**Review by thewtex — CHANGES_REQUESTED** [review]
@tbirdso awesome!! :rocket: :man_singer: 

I will create an issue, but in another PR, we should do a little reorganizing and create a separate package out of the pure python code, `hasi` that has `itk-hasi` as a dependency.

A few minor issues noted inline.

**Comment by tbirdso** [comment]
@thewtex Marking as draft again for now to add pytest to CI checks.

**Comment by tbirdso** [comment]
@thewtex Marked as ready for review again. Note that this PR will add CI `build-test-python` checks for Python modules with pytest. Existing check failures are tracked in [Issue 30](https://github.com/KitwareMedical/HASI/issues/30) and predate these changes.

**Review by thewtex — APPROVED** [review]
@tbirdso awesome!!

**Pull request merged**